src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -1001,12 +1001,19 @@ public void Sort(IComparer<T> comparer)
             }
         }
 
-        // ToArray returns a new Object array containing the contents of the List.
+        // ToArray returns an array containing the contents of the List.
         // This requires copying the List, which is an O(n) operation.
         public T[] ToArray() {
             Contract.Ensures(Contract.Result<T[]>() != null);
             Contract.Ensures(Contract.Result<T[]>().Length == Count);
 
+#if FEATURE_CORECLR
+            if (_size == 0)
+            {
+                return Array.Empty<T>();
+            }
+#endif
+
             T[] array = new T[_size];
```

Comment:
Agree there would be some extra work here in copy.  Feel like more `Array.Empty<T>` overall would get more use if we have a clean, simple way to have allocations return it automatically.